src/main/java/org/myrobotlab/service/ProgramAB.java

```diff
@@ -129,13 +104,98 @@ public String toString() {
    */
   public ProgramAB(String n, String id) {
     super(n, id);
-    availableBots = getBots();
+    // botPaths = initBotPaths();

```

Comment:
I feel like this should be done in the startService.. not in the constructor...

Reply:
agreed ... moved